components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/java/org/wso2/carbon/mediator/enrich/ui/EnrichMediator.java

```diff
@@ -177,13 +178,23 @@ public OMElement serialize(OMElement parent) {
 
             if (sourceInlineXML != null && !sourceInlineXML.equals("")) {
                 OMElement inlineXMLElement;
+                JsonParser parser = new JsonParser();
+                // check for inline JSON values
                 try {
-                    inlineXMLElement = AXIOMUtil.stringToOM(sourceInlineXML);
-                    if (inlineXMLElement != null) {
-                        sourceElem.addChild(inlineXMLElement);
+                    parser.parse(sourceInlineXML);
+                    String formattedString = "<source xmlns=\"" + SynapseConstants.SYNAPSE_NAMESPACE +
+                            "\" type=\"inline\">" + sourceInlineXML + "</source>";
+                    sourceElem = AXIOMUtil.stringToOM(formattedString);
+                } catch (Exception ex) {
```

Comment:
This will catch `RuntimeExceptions` as well. Isn't it enough to catch `XMLStreamException` here?

Reply:
Writing core logic using try catch is unreadable and expensive. Isn't it? Can't we have a validation and avoid this?

Reply:
This is the method I found to validate the string is a JSON. Can you suggest a better approach? (maybe using a regex)

Reply:
discussed to stick with the current approach